core/src/mindustry/world/modules/ItemModule.java

```diff
@@ -66,11 +66,28 @@ public int total(){
         return total;
     }
 
+    // Making the take() loop persistent so it does not return the same item twice in a row unless there is nothing else to return.
+    protected int takeRotation;
```

Comment:
Instance variables should always be declared at the top of a class.

Reply:
Of course, sorry about that.

Just wanted to put the tweak out for people to try and hopefully see the modified behavior or something resembling it make it into a future release.